staging/src/k8s.io/kubectl/pkg/cmd/run/run.go

```diff
@@ -166,6 +166,23 @@ func NewCmdRun(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Co
 	addRunFlags(cmd, o)
 	cmdutil.AddApplyAnnotationFlags(cmd)
 	cmdutil.AddPodRunningTimeoutFlag(cmd, defaultPodAttachTimeout)
+
+	// Deprecate the cascade flag. If set, it has no practical effect since the created pod has no dependents.
+	// TODO: Remove the cascade flag from the run command in kubectl 1.29
+	cmd.Flags().MarkDeprecated("cascade", "This flag will be removed in version 1.29")
+
+	// Deprecate and hide unused flags.
+	// These flags are being added to the run command by DeleteFlags to support pod deletion after attach,
+	// but they are not used if set, so they effectively do nothing.
+	// TODO: Remove these flags from the run command in kubectl 1.29
+	cmd.Flags().MarkDeprecated("filename", "This flag is ignored and will be removed in version 1.29")
+	cmd.Flags().MarkDeprecated("force", "This flag is ignored and will be removed in version 1.29")
+	cmd.Flags().MarkDeprecated("grace-period", "This flag is ignored and will be removed in version 1.29")
+	cmd.Flags().MarkDeprecated("kustomize", "This flag is ignored and will be removed in version 1.29")
+	cmd.Flags().MarkDeprecated("recursive", "This flag is ignored and will be removed in version 1.29")
+	cmd.Flags().MarkDeprecated("timeout", "This flag is ignored and will be removed in version 1.29")
+	cmd.Flags().MarkDeprecated("wait", "This flag is ignored and will be removed in version 1.29")
```

Comment:
Nit: I built this PR to see how the message actually looks, and apparently it is concatenated to the standard message with a comma:
```
Flag --filename has been deprecated, This flag is ignored and will be removed in version 1.29
Flag --kustomize has been deprecated, This flag is ignored and will be removed in version 1.29
```

Suggestion: `Flag --kustomize has been deprecated, because it is not used by this command. It will be removed in version 1.29`

Reply:
Yeah, I thought that was a little weird too.  I was following how most of the other deprecation messages were worded, but maybe it is a good time to break from that and try to make it better.  I'll update the messages.

Reply:
Updated:
```
Flag --filename has been deprecated, because it is not used by this command. It will be removed in version 1.29.
Flag --kustomize has been deprecated, because it is not used by this command. It will be removed in version 1.29.
Flag --cascade has been deprecated, because it is not relevant for this command. It will be removed in version 1.29.
```

Hopefully it will be clear these flags are only going to be removed from THIS command, and not every command.  I think it should be clear enough.

Reply:
👌